components/eam/src/physics/crm/vertical_diffusion.F90

```diff
@@ -132,1298 +106,765 @@ module vertical_diffusion
+      call cnst_get_ind( 'NUMICE', ixnumice )
+  endif
+
+  ! prog_modal_aero determines whether prognostic modal aerosols are present in the run.
+  call phys_getopts(prog_modal_aero_out=prog_modal_aero)
+  if (prog_modal_aero) then
+
+     ! Get the constituent indices of the number and mass mixing ratios of the modal 
+     ! aerosols.
+     !
+     ! N.B. - This implementation assumes that the prognostic modal aerosols are 
+     !        impacting the climate calculation (i.e., can get info from list 0).
+     ! 
+
+     ! First need total number of mam constituents
+     call rad_cnst_get_info(0, nmodes=nmodes)
+     do m = 1, nmodes
+        call rad_cnst_get_info(0, m, nspec=nspec)
+        pmam_ncnst = pmam_ncnst + 1 + nspec
+     end do
+
+     allocate(pmam_cnst_idx(pmam_ncnst))
+
+     ! Get the constituent indicies
+     im = 1
+     do m = 1, nmodes
+        call rad_cnst_get_mode_num_idx(m, pmam_cnst_idx(im))
+        im = im + 1
+        call rad_cnst_get_info(0, m, nspec=nspec)
+        do l = 1, nspec
+           call rad_cnst_get_mam_mmr_idx(m, l, pmam_cnst_idx(im))
+           im = im + 1
+        end do
+     end do
+  end if
+
+  !-----------------------------------------------------------------------------
+  ! Initialize molecular diffusivity module                                    
+  ! Note that computing molecular diffusivities is a trivial expense, but constituent
+  ! diffusivities depend on their molecular weights. Decomposing the diffusion matric
+  ! for each constituent is a needless expense unless the diffusivity is significant.
+  !-----------------------------------------------------------------------------
+
+  !-----------------------------------------------------------------------------
+  ! Initialize molecular diffusion and get top and bottom limits
+  !-----------------------------------------------------------------------------
+  if( do_molec_diff ) then
+     call init_molec_diff( r8, pcnst, rair, mwdry, avogad, gravit, cpair, boltz, errstring)
+
+     call handle_errmsg(errstring, subname="init_molec_diff")
+
+     call addfld( 'TTPXMLC', horiz_only, 'A', 'K/S', 'Top interf. temp. flux: molec. viscosity' )
+     call add_default ( 'TTPXMLC', 1, ' ' )
+     if( masterproc ) write(iulog,fmt='(a,i3,5x,a,i3)') 'NTOP_MOLEC =', ntop_molec, 'NBOT_MOLEC =', nbot_molec
+  end if
+
+  !-----------------------------------------------------------------------------
+  ! Initialize eddy diffusivity module
+  !-----------------------------------------------------------------------------
+  ! ntop_eddy must be 1 or <= nbot_molec (it is always 1 except for WACCM-X)
+  ntop_eddy = 1
+  nbot_eddy = pver
+
+  if (masterproc) write(iulog, fmt='(a,i3,5x,a,i3)') 'NTOP_EDDY  =', ntop_eddy, 'NBOT_EDDY  =', nbot_eddy
+  if (masterproc) write(iulog,*) 'vertical_diffusion_init: eddy_diffusivity scheme:  Holtslag and Boville'
+  
+  call init_hb_diff(gravit, cpair, ntop_eddy, nbot_eddy, pref_mid, karman)
```

Comment:
In this file, many constants defined in physconst module are passed to the subroutine in the argument list. I guess this is just one's coding style. I don't think it's necessary but let's just leave it at that.

Reply:
This is just taken from how it was previously called in the section of E3SM from CAM3. We can change it later if we have a good reason, but I'd like to leave it for consistency.